trailmap/acquisition_paths.py

```diff
@@ -0,0 +1,42 @@
+import networkx as nx
+from pprint import pprint
+
+
+def conduct_apa(facility_dict_in, facility_dict_out):
+    ''' '''
+    G = build_graph(facility_dict_in, facility_dict_out)
+
+    sources = list(node for node, in_deg in G.in_degree() if in_deg == 0)
+    sinks = list(node for node, out_deg in G.out_degree() if out_deg == 0)
+
+    print("\nSimple paths")
+    pathways = find_simple_paths(G, sources, sinks)
+    print(pathways)
+
+    return G, pathways
+
+
+def build_graph(facility_dict_in, facility_dict_out):
+    '''Builds NetworkX graph from Cyclus input file
+    '''
+    G = nx.MultiDiGraph()
+    G.add_nodes_from(facility_dict_in.keys())
+    for receiver, incommods in facility_dict_in.items():
+        # find all facilities with that commod as an outcommod
+        for sender, outcommods in facility_dict_out.items():
+            connection = list(set(incommods).intersection(outcommods))
+            for commod in connection:
+                G.add_edge(sender, receiver, commodity=commod)
+    return G
+
+
+def find_simple_paths(G, sources, sinks):
+    ''' finds all simple paths between a given list of sources and targets
+    '''
+    pathways = set()
+    for source, sink in [(x, y) for x in sources for y in sinks]:
+        paths = list(nx.all_simple_paths(G, source=source, target=sink))
+        for path in paths:
+            pathways.add(tuple(path))
```

Comment:
maybe you can inline this one ?

Reply:
I actually had this as a single line in an older/first version
`[pathways.update(tuple(path)) for path in paths]`
@gonuke recommended I break it into multi-line for clarity

Reply:
I just found it silly to loop over the `paths` for append each element to the pathways.

Maybe you could convert paths into a tuple and append that topple directly ?

Reply:
I don't want a tuple of paths in pathways though.
`list(nx.all_simple_paths(G, source=source, target=sink))` returns a list of lists, for example 
`[['SourceA', 'SinkA'], ['SourceA', 'Facility', 'SinkA']]. I want to turn each individual element in that list into a tuple and add it to pathways set

Let's say `pathways = {('SourceB', 'SinkB')}` and now we're iterating over source=SourceA and sink=SinkA. The output of `list(nx.all_simple_paths(G, source=source, target=sink))` is `[['SourceA', 'SinkA'], ['SourceA', 'Facility', 'SinkA']]`, a list of lists. Turning that into a tuple would give `(['SourceA', 'SinkA'], ['SourceA', 'Facility', 'SinkA'])`, which isn't what I want, and can't be added anyway because it still includes an unhashable type. I want each element to become a tuple so it can be added to the set of tuples `pathways`.

In this mini-example, what I want to reach is {('SourceB', 'SinkB'), ('SourceA', 'SinkA'), ('SourceA', 'Facility', 'SinkA')}.

I don't really have a preference between `[pathways.update(tuple(path)) for path in paths]` and 
```for path in paths:
            pathways.add(tuple(path))```
but I also don't see a clear other way to add elements from a list of lists as individual tuples to a set

Reply:
```
pathways.add({ tuple(path) for path in nx.all_simple_paths(...)})
```